ee/app/livechat-enterprise/server/lib/LivechatEnterprise.js

```diff
@@ -113,4 +114,51 @@ export const LivechatEnterprise = {
 
 		return LivechatTag.createOrUpdateTag(_id, tagData, tagDepartments);
 	},
+
+	savePriority(_id, priorityData) {
+		check(_id, Match.Maybe(String));
+
+		check(priorityData, {
+			name: String,
+			description: Match.Optional(String),
+			color: String,
+			dueTimeInMinutes: String,
+		});
+		LivechatRooms.updatePriorityDataByPriorityId(_id, priorityData);
+		Subscriptions.updatePriorityDataByPriorityId(_id, priorityData);
+		LivechatInquiry.updatePriorityDataByPriorityId(_id, priorityData);
+		return LivechatPriority.createOrUpdatePriority(_id, priorityData);
+	},
+
+	removePriority(_id) {
+		check(_id, String);
+
+		const priority = LivechatPriority.findOneById(_id, { fields: { _id: 1 } });
+
+		if (!priority) {
+			throw new Meteor.Error('priority-not-found', 'Priority not found', { method: 'livechat:removePriority' });
+		}
+		LivechatRooms.unsetPriorityByPriorityId(_id);
+		Subscriptions.unsetPriorityByPriorityId(_id);
+		LivechatInquiry.unsetPriorityByPriorityId(_id);
+		return LivechatPriority.removeById(_id);
+	},
+
+	savePriorityDataOnRooms(room, user) {
+		if (room.omnichannel && room.omnichannel.priority) {
+			const priority = LivechatPriority.findOneById(room.omnichannel.priority._id);
+			Subscriptions.setPriorityByRoomId(room._id, priority);
+			LivechatInquiry.setPriorityByRoomId(room._id, priority);
+		} else {
+			Subscriptions.unsetPriorityByRoomId(room._id);
+			LivechatInquiry.unsetPriorityByRoomId(room._id);
+		}
+		const extraData = {
+			priorityData: {
+				definedBy: user,
+				priority: room.omnichannel && room.omnichannel.priority ? room.omnichannel.priority : {},
```

Comment:
I would destructure the `omnichannel` object before assing the `priority` property.